build-aux/cargo_build.py

```diff
@@ -22,7 +23,16 @@
     output_file: {output_file}
 """, file=sys.stderr)
 
-cargo_call = f"env {cargo_env} {cargo_cmd} build {cargo_options}"
+# Detecting architecture
+if sys.platform == 'win32':
+    if platform.machine().lower() == 'arm64':
+        rustflags = '-C linker=clang'
+    else:
+        rustflags = ''
+else:
+    rustflags = ''
+
+cargo_call = f"env {cargo_env} RUSTFLAGS='{rustflags}' {cargo_cmd} build {cargo_options}"
```

Comment:
Thank you for the feedback about moving the logic to the meson build file. I want to be transparent - I've been using AI to help me adjust the code as I have limited experience with Rust and the app building process. This approach has proven to be quite inefficient, and despite multiple attempts, I haven't been able to successfully move the logic to the meson build file.

Rather than continue with AI-assisted code adjustments that might not meet the project's standards, I think I can be more helpful by offering to test any changes you implement on my ARM64 device (Surface Pro 11). This would ensure the changes work correctly on ARM64 Windows hardware, which might be valuable for the project.

Let me know if you'd like me to test any changes you make, and I'll be happy to provide detailed feedback about how it works on ARM64.